app/serializers/search_and_compare/course_serializer.rb

```diff
@@ -41,7 +41,7 @@ class CourseSerializer < ActiveModel::Serializer
     attribute(:Name)                                  { object.name }
     attribute(:ProgrammeCode)                         { object.course_code }
     # using server date time not utc, so it's local date time?
-    attribute(:StartDate)                             { object.start_date.utc.strftime('%Y-%m-%dT%H:%M:%S') }
+    attribute(:StartDate)                             { object.start_date&.utc&.strftime('%Y-%m-%dT%H:%M:%S') }
```

Comment:
Why would `.utc` return `nil`? I don't think you need that second guard. And does this return the right thing when `start_date` is nil? I'm guessing 'null'?

Reply:
I think I was partly confused and forgot when commenting that you need `&.` after `utc` since otherwise it'll try to run `nil.utc` if `objects.start_date` is nil. I guess this is one of the reasons `&.` is considered an anti-pattern by some.